pytest_catchlog.py

```diff
@@ -71,15 +71,22 @@ def catching_logs(handler, filter=None, formatter=None,
                 yield handler
 
 
-def add_option_ini(parser, option, dest, default=None, help=None):
+def add_option_ini(parser, option, dest, action=None, default=None, help=None):
     parser.addini(dest, default=default,
                   help='default value for ' + option)
-    parser.getgroup('catchlog').addoption(option, dest=dest, help=help)
+    parser.getgroup('catchlog').addoption(option,
+                                          action=action,
+                                          dest=dest,
+                                          default=default,
+                                          help=help)
 
-def get_option_ini(config, name):
+
+def get_option_ini(config, name, default=None):
     ret = config.getoption(name)  # 'default' arg won't work as expected
     if ret is None:
         ret = config.getini(name)
+    if ret == '':
+        return default
```

Comment:
Also, if we don't pass the default as `None` to the option parser, the value of `ret = config.getoption(name)` will be `True` probably because the `action='store_false'` tell's it that the default should be `False` somewhere... And again, we wouldn't get past that value to investigate what the user has in the ini file.